src/input/type_record.cc

```diff
@@ -180,7 +181,7 @@ Record &Record::derive_from(AbstractRecord &parent) {
 	data_->parent_ptr_.push_back( boost::make_shared<AbstractRecord>(parent) );
 
 	if (data_->keys.size() == 0) {
-    	data_->declare_key("TYPE", boost::shared_ptr<TypeBase>(NULL), Default::obligatory(), "Sub-record Selection.");
+    	data_->declare_key("TYPE", boost::shared_ptr<TypeBase>(NULL), Default( type_name() ), "Sub-record Selection.");
```

Comment:
Why this change? I 'm suspicious that this can allow missing TYPE key in abstract records.
Why declare_type_key can not be used?